### DISCUSSION TITLE
Question:  Why is init_canonical loaded via its own auto-loader?

### DISCUSSION THREAD

**Discussion by lat9** [zencart/zencart#5790] [Q&A]
Not a bug, per se, but a question.  Why is the **core** file `init_canonical.php` loaded with its own auto-loader (`config.canonical.php`) instead of being loaded as part of the other **core** initializations in `config.core.php`?

It just feels like that could have the possibility of missing a prerequisite for that file's loading, since it's separated out.

**Comment by scottcwilson** [comment]
@zcwilt Do you know the history on this one?

**Comment by drbyte** [comment]
I believe we did it that way to make adding it simpler for those doing upgrades because they only needed to add the files, not edit existing files.
